### ISSUE TITLE
Morris FAB rules: must be on last order and must be voided at line level, not at order level

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2222]
_Imported from trac ticket 2222.
Created by **dwarren**
Opened in trac: Mar 14, 2013, 12:37:02 PM
Last modified in trac: Apr 12, 2013, 3:20:23 PM_
***
_by dwarren:_
 * FAB products must be on the last (to deliver) open order
 * FAB products must be voided at the line level (not as part of entire order void, unless there is no other order at all)

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 15, 2013, 6:00:38 PM:_

(Changeset [[3918]](../commit/4c9290e403ac637f95992abfe6eb69d668868bc3))
* Morris FAB rules: must be on last order and must be voided at line level, not at order level (closes #2222)

**Comment by kentbower** [comment]
_**kbower** trac status change on Mar 15, 2013, 6:00:38 PM:_ **closed**

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Apr 11, 2013, 5:23:03 PM:_

We had a rule created to move the Protection coverage SKU to the latest delivery date within an order when splitting lines.
We discovered in training a little flaw in the rule:
1.       It was prompting the user to move the protection to the latest delivery date that didn’t have any covered items on it.  It seems it is just looking at the delivery date and not what items are covered.  The coverage should be on the last delivery date of the items being covered by the protection.
2.       There should be an exception added  to this rule that we forgot about
       a. …if any covered items are on a split order and the delivery zone is XAL, it should be put on that split XAL order.

**Comment by kentbower** [comment]
_**sjohnson** trac status change on Apr 11, 2013, 5:23:03 PM:_ **reopened**

**Comment by kentbower** [comment]
_**kbower** trac comment on Apr 12, 2013, 8:42:42 AM:_

Regarding the changes requested [below](comment:3), is it sufficient that any and _all protection SKUs_ all be forced to the last order with any item that is covered by _some_ protection?

This is less effort, but if you often sell more than one protection sku, then I can change it by adding the logic to make sure each is on the last order with items covered by it specifically.

In other words, is it sufficient without matching coverage to protection skus individually, or do you often sell more than one protection plan and so it should be smart enough to ensure the protection sku is on a ticket with items it covers?

Kent

**Comment by kentbower** [comment]
_**kbower** trac comment on Apr 12, 2013, 3:20:23 PM:_

(Changeset [[4024]](../commit/44aeddae973f31a702201299d572ddc0aeb6c4df))
* closes #2222, changes to Morris FAB on last order rule (mor)

**Comment by kentbower** [comment]
_**kbower** trac status change on Apr 12, 2013, 3:20:23 PM:_ **closed**
